src/table/table.js

```diff
@@ -27,26 +40,45 @@ export default class Table extends React.Component<TablePropsT> {
   };
 
   render() {
+    const {isLoading, columns, data, horizontalScrollWidth} = this.props;
     return (
       <StyledTable
         data-baseweb="table"
-        aria-colcount={this.props.columns.length}
-        aria-rowcount={this.props.data.length}
+        aria-colcount={columns.length}
+        aria-rowcount={data.length}
       >
-        <StyledHead $width={this.props.horizontalScrollWidth}>
-          {this.props.columns.map((column, index) => (
+        {isLoading && (
+          <ProgressBar
+            infinite
+            overrides={{
+              Bar: {
+                style: {
+                  marginBottom: 0,
+                  marginLeft: 0,
+                  marginRight: 0,
+                  marginTop: 0,
+                },
+              },
+            }}
+          />
+        )}
+        <StyledHead $width={horizontalScrollWidth}>
+          {columns.map((column, index) => (
             <StyledHeadCell key={index}>{column}</StyledHeadCell>
           ))}
         </StyledHead>
-
-        <StyledBody $width={this.props.horizontalScrollWidth}>
-          {this.props.data.map((row, index) => (
-            <StyledRow key={index}>
-              {row.map((cell, cellIndex) => (
-                <StyledCell key={cellIndex}>{cell}</StyledCell>
-              ))}
-            </StyledRow>
-          ))}
+        <StyledBody $width={horizontalScrollWidth}>
+          {data.length > 0 ? (
+            data.map((row, index) => (
+              <StyledRow key={index}>
+                {row.map((cell, cellIndex) => (
+                  <StyledCell key={cellIndex}>{cell}</StyledCell>
+                ))}
+              </StyledRow>
+            ))
+          ) : (
+            <EmptyWrapper>{isLoading ? 'Loading...' : 'No data'}</EmptyWrapper>
```

Comment:
Please remove this message, along with the `EmptyWrapper` component. It will unexpectedly change existing applications. Maybe it could be added in a separate pr where the feature is opt-in